persistence/trees/trees_test.go

```diff
@@ -1,21 +1,201 @@
-package trees
+package trees_test
 
-import "testing"
+import (
+	"encoding/hex"
+	"log"
+	"math/big"
+	"testing"
+
+	"github.com/pokt-network/pocket/persistence"
+	"github.com/pokt-network/pocket/persistence/trees"
+	"github.com/pokt-network/pocket/runtime"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/test_artifacts"
+	"github.com/pokt-network/pocket/runtime/test_artifacts/keygen"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/utils"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	defaultChains          = []string{"0001"}
+	defaultStakeBig        = big.NewInt(1000000000000000)
+	defaultStake           = utils.BigIntToString(defaultStakeBig)
+	defaultStakeStatus     = int32(coreTypes.StakeStatus_Staked)
+	defaultPauseHeight     = int64(-1) // pauseHeight=-1 implies not paused
+	defaultUnstakingHeight = int64(-1) // unstakingHeight=-1 implies not unstaking
+
+	testSchema = "test_schema"
+
+	genesisStateNumValidators   = 5
+	genesisStateNumServicers    = 1
+	genesisStateNumApplications = 1
+)
+
+const (
+	h1 = "5282ee91a3ec0a6f2b30e4780b369bae78c80ef3ea40587fef6ae263bf41f244"
+)
 
-// TECHDEBT(#836): Tests added in https://github.com/pokt-network/pocket/pull/836
 func TestTreeStore_Update(t *testing.T) {
-	// TODO: Write test case for the Update method
-	t.Skip("TODO: Write test case for Update method")
+	pool, resource, dbUrl := test_artifacts.SetupPostgresDocker()
+	t.Cleanup(func() {
+		require.NoError(t, pool.Purge(resource))
+	})
+
+	t.Run("should update actor trees, commit, and modify the state hash", func(t *testing.T) {
+		pmod := newTestPersistenceModule(t, dbUrl)
+		context := newTestPostgresContext(t, 0, pmod)
+
+		require.NoError(t, context.SetSavePoint())
+
+		hash1, err := context.ComputeStateHash()
+		require.NoError(t, err)
+		require.NotEmpty(t, hash1)
+		require.Equal(t, hash1, h1)
+
+		_, err = createAndInsertDefaultTestApp(t, context)
+		require.NoError(t, err)
+
+		require.NoError(t, context.SetSavePoint())
+
+		hash2, err := context.ComputeStateHash()
+		require.NoError(t, err)
+		require.NotEmpty(t, hash2)
+		require.NotEqual(t, hash1, hash2)
+	})
+
+	t.Run("should fail to rollback when no treestore savepoint is set", func(t *testing.T) {
+		pmod := newTestPersistenceModule(t, dbUrl)
+		context := newTestPostgresContext(t, 0, pmod)
+
```

Comment:
Can you update this test to:

1. Insert the test app
2. not create a savepoint (as exists)
3. Verify hash is unchanged after the failed rollback?

Reply:
Addressed in #944 to unblock this PR for now.